app/controllers/support.js

```diff
@@ -0,0 +1,25 @@
+import Controller from '@ember/controller';
+import { tracked } from '@glimmer/tracking';
+
+const SUPPORTS = [
+  {
+    inquire: 'crate-violation',
+    label: 'Report a crate that violates policies',
+  },
+];
+
+const VALID_INQUIRE = new Set(SUPPORTS.map(s => s.inquire));
+
+export default class SupportController extends Controller {
+  queryParams = ['inquire', 'name'];
+
+  @tracked inquire;
```

Comment:
since this is in a controller I think the property keeps its value across page navigations. in other words: if you press the "report crate" button the property will be set, if you navigate away from the page, then you press the support link in the footer, you will not get to the overview state, but to the "report crate" state again. I'm not 100% sure about this, but we should probably add a test to ensure that navigating to `/support` without query params resets the state.

Reply:
One more thing I noticed is that, if you're on the support page with query, if the support link in footer is defined as `<LinkTo @route="support" />` it will still inherit those query params. There doesn't seem to be a simple way to avoid this inheritance in the template. You need to use hash to manual overwrite all query params.

Reply:
I'm wondering if there's a way to test this without `<LinkTo @route="support"/>`. We already have `<LinkTo @route="support" @query={{hash inquire=null crate=null }}>Support</LinkTo>` that reset the query params by setting their values to null.

Reply:
not sure about playwright, but in the Ember.js test suite `visit('/support')` might be sufficient as it does not reload the whole app

Reply:
Hmm, I just commented out `resetController` and expected the following to fail, but it didn't 😅 .
```js
  test('should not retain query params when exiting and then returning', async function (assert) {
    await visit('/support?inquire=crate-violation');
    assert.strictEqual(currentURL(), '/support?inquire=crate-violation');
    assert
      .dom('[data-test-id="support-main-content"] section')
      .exists({ count: 1 })
      .hasAttribute('data-test-id', 'crate-violation-section');

    await visit('/');
    assert.strictEqual(currentURL(), '/');
    await visit('/support');
    assert.strictEqual(currentURL(), '/support');
    assert
      .dom('[data-test-id="support-main-content"] section')
      .exists({ count: 1 })
      .hasAttribute('data-test-id', 'inquire-list-section');
  });

```
However, if there's a `<LinkTo @route="support">`, I can see that the generated URL has sticky query params.

Reply:
interesting, maybe I was wrong then :D

Reply:
I figured out how to test this in qunit but not playwright. I don't know how to mock the route in playwright QQ.